### PR TITLE
refactor: Add sources filtering by provider

### PR DESCRIPTION
**Pull request by adiabramovitch**
This is a temporary fix until @amirfefer will finish adding the GCP and Azure UI part.

### CODE REVIEW

File: src/Components/DescriptionListAWS/index.js [hunk 1]
```diff
@@ -7,9 +7,10 @@ import { SOURCES_QUERY_KEY } from '../../API/queryKeys';
 import { fetchSourcesList } from '../../API';
 import { useWizardContext } from '../Common/WizardContext';
 
+const PROVIDER = 'aws';
 const DescriptionListAWS = ({ imageName }) => {
   const [wizardContext] = useWizardContext();
-  const { error, data: sources } = useQuery(SOURCES_QUERY_KEY, fetchSourcesList);
+  const { error, data: sources } = useQuery(SOURCES_QUERY_KEY, fetchSourcesList(PROVIDER));
```
Comment (ezr-ondrej) [inline src/Components/DescriptionListAWS/index.js:13]: I don't think we necessarily need to filter here :)

File: src/Components/SourcesSelect/index.js [hunk 2]
```diff
@@ -20,7 +21,7 @@ const SourcesSelect = ({ setValidation }) => {
     error,
     isLoading,
     data: sources,
-  } = useQuery(SOURCES_QUERY_KEY, fetchSourcesList, {
+  } = useQuery(SOURCES_QUERY_KEY, fetchSourcesList(PROVIDER), {
```
Comment (ezr-ondrej) [inline src/Components/SourcesSelect/index.js:24]: we need to pass a function, not it's result here
```suggestion
  } = useQuery(SOURCES_QUERY_KEY, () => fetchSourcesList(PROVIDER), {
```
Or another solution would be to change `fetchSourcesList` to return a function

**Review by ezr-ondrej — CHANGES_REQUESTED** [review]

File: src/Components/DescriptionListAWS/index.js [hunk 3]
```diff
@@ -7,9 +7,10 @@ import { SOURCES_QUERY_KEY } from '../../API/queryKeys';
 import { fetchSourcesList } from '../../API';
 import { useWizardContext } from '../Common/WizardContext';
 
+const PROVIDER = 'aws';
```
Comment (amirfefer) [inline src/Components/DescriptionListAWS/index.js:10]: with  #195 the `provider` is given by IB and stored in the wizard context

**Review by amirfefer — COMMENTED** [review]
Thanks @adiabramovitch 👍 
a rebase with #195 will  give you the provider with no extra constants

In second thought, I've refacored the final descriptation page for general provider, so rebased would be quicker doing the rebase in #195 

Tested with aws and gce sources, works as expected

**Review by amirfefer — APPROVED** [review]

**Pull request merged**